cmd/ubuntu-image/main.go

```diff
@@ -0,0 +1,163 @@
+where <volume> is the volume name taken from the gadget.yaml file.
+Use this option instead of the deprecated -o/--output option.`
+	debugFlagDesc := `Enable debugging output`
+
+	// state machine flags
+	directoryFlagDesc := `The working directory in which to download and unpack all the
+source files for the image. This directory can exist or not, and it is not
+removed after this program exits. If not given, a temporary working
+directory is used instead, which *is* deleted after this program exits. Use -w
+if you want to be able to resume a partial state machine run.`
+	untilFlagDesc := `Run the state machine until the given STEP, non-inclusively.
+STEP can be a name or number.`
+	thruFlagDesc := `Run the state machine through the given STEP, inclusively.
+STEP can be a name or number.`
+	resumeFlagDesc := `Continue the state machine from the previously saved state.
+It is an error if there is no previous state.`
+
+	// create the snap command and add snap specific flags to it
+	snap := &cobra.Command{
+		Use:   "snap [model_assertion]",
+		Short: "Create snap-based Ubuntu Core image.",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println("ubuntu-image snap functionality to be added later")
+		},
+	}
+
+	var Snap, Channel string
+	var ConsoleConf bool
+	snap.PersistentFlags().StringVarP(&Snap, "snap", "", "", snapFlagDesc)
+	snap.PersistentFlags().StringVarP(&Channel, "channel", "c", "", channelFlagDesc)
+	snap.PersistentFlags().BoolVarP(&ConsoleConf, "disable-console-conf", "", false, consoleConfFlagDesc)
+
+	// create the classic command and add classic specific flags to it
+	classic := &cobra.Command{
+		Use:   "classic [graph_tree]",
+		Short: "Create debian-based Ubuntu Classic image.",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println("ubuntu-image classic functionality to be added later")
+		},
+	}
+
+	var Project, Filesystem, Suite, Architecture, Subproject, Subarchitecture, ExtraPpas string
+	var WithProposed bool
+
+	classic.PersistentFlags().StringVarP(&Project, "project", "p", "", projFlagDesc)
+	classic.PersistentFlags().StringVarP(&Filesystem, "filesystem", "f", "", filesystemFlagDesc)
+	classic.PersistentFlags().StringVarP(&Suite, "suite", "s", "", distributionFlagDesc)
+	classic.PersistentFlags().StringVarP(&Architecture, "arch", "a", "", architectureFlagDesc)
+	classic.PersistentFlags().StringVarP(&Subproject, "subproject", "", "", subprojFlagDesc)
+	classic.PersistentFlags().StringVarP(&Subarchitecture, "subarch", "", "", subarchFlagDesc)
+	classic.PersistentFlags().StringVarP(&ExtraPpas, "extra-ppas", "", "", ppasFlagDesc)
+	classic.PersistentFlags().BoolVarP(&WithProposed, "with-proposed", "", false, withProposedFlagDesc)
+
+	// add common flags to both commands
+	var Size, FileList, CloudInit, HooksDirectory, DiskInfo, Output string
+	var Debug bool
+	snap.PersistentFlags().StringVarP(&Size, "image-size", "i", "", sizeFlagDesc)
```

Comment:
Is there a way to remove the redundancy of these blocks adding the common flags to both commands?

Reply:
Investigating using the pflags library from spf13 to do this is on my todo list.